src/btree/bt_random.c

```diff
@@ -302,11 +297,15 @@ __wt_btcur_next_random(WT_CURSOR_BTREE *cbt)
 	WT_UPDATE *upd;
 	wt_off_t size;
 	uint64_t n, skip;
+	uint32_t read_flags;
 	bool valid;
 
 	btree = cbt->btree;
 	cursor = &cbt->iface;
 	session = (WT_SESSION_IMPL *)cbt->iface.session;
+	read_flags = 0;
+	if (F_ISSET(cbt, WT_CBT_READ_WONT_NEED))
+		FLD_SET(read_flags, WT_CBT_READ_WONT_NEED);
```

Comment:
Two things:
1. I think `read_flags` needs to be initialized to `WT_READ_RESTART_OK`.
2. I think you need to set `WT_READ_WONT_NEED` into `read_flags` (not the CBT value).

Reply:
Correct on both accounts, thanks for the catch!